lib/AST/ASTWalker.cpp

```diff
@@ -330,6 +330,15 @@ class Traversal : public ASTVisitor<Traversal, Expr*, Stmt*,
       }
     }
 
+    // FIXME: FIXME_NOW: DETERMINE: Is this or something like it needed?  Right
+    //        now it's causing assertion failures during AST verification.
+    //     
+    //     if (AFD->hasSingleExpressionBody()) {
+    //       if (Expr *body = doIt(AFD->getSingleExpressionBody()))
+    //         AFD->setSingleExpressionBody(body);
+    //       else
+    //         return true;
+    //     }
```

Comment:
Maybe continue processing the body the old way, but check if it goes from single-expression to non-single-expression? I think PlaygroundTransform might do that (because it will insert logging).

Reply:
Is this the kind of thing that you had in mind?

```
    // Check whether a function which was single-expression is no longer because
    // of code being inserted.
    if (AFD->hasSingleExpressionBody()) {
      bool stillHasSingleExpressionBody = true;
      if (auto *BS = AFD->getBody()) {
        switch (BS->getNumElements()) {
          case 0: {
            stillHasSingleExpressionBody = false;
            break;
          }
          case 1: {
            auto element = BS->getElement(0);
            if (auto stmt = element.dyn_cast<Stmt *>()) {
              if (!isa<ReturnStmt>(stmt) && !isa<FailStmt>(stmt)) {
                stillHasSingleExpressionBody = false;
                break;
              }
            } else if (element.is<Decl *>()) {
              stillHasSingleExpressionBody = false;
            }
            break;
          }
          case 2: {
            if (!BS->getElement(0).is<Expr *>()) {
              stillHasSingleExpressionBody = false;
              break;
            }
            if (auto *stmt = BS->getElement(1).dyn_cast<Stmt *>()) {
              if (!isa<ReturnStmt>(stmt)) {
                stillHasSingleExpressionBody = false;
              }
              break;
            } else {
              stillHasSingleExpressionBody = false;
              break;
            }
          }
        }
      }
      if (!stillHasSingleExpressionBody) {
        AFD->setHasSingleExpressionBody(false);
      }
    }
```

Reply:
:-( Yeah, this is pretty nasty. I don't know what to do here. 

(I do suggest writing PlaygroundTransform tests and seeing how they handle all this.)

Reply:
Okay, several tests added here: 
https://github.com/apple/swift/pull/23251/commits/a7daf25f74f42b49414e459bb5b45dc8630902dd .  They look to behave correctly, so maybe we don't need to do anything here after all?